### PR TITLE
#50 | add the test reporter. 

### PR DESCRIPTION
**Pull request by skayikci**
Testing locally using ["act"](https://github.com/nektos/act)

install github cli
install act
use your github token and run this command:

`act -s GITHUB_TOKEN=YOUR_TOKEN_HERE --container-architecture linux/amd64`

ask me if you have any issues. since it might be troublesome to test it locally.

### CODE REVIEW

**Comment by skayikci** [comment]
i'll need to get the deployment information in order to create the deployment scripts.

**Comment by skayikci** [comment]
@agitrubard @Rapter1990 I'll need to try it out in order to test. i might need to reopen and merge for couple of times. FYI.

**Comment by skayikci** [comment]
i've tested following on branch PRs:

- [X]  happy path: https://github.com/afet-yonetim-sistemi/ays-be/actions/runs/5405658941

- [X] break a test on branch: https://github.com/afet-yonetim-sistemi/ays-be/actions/runs/5405677154
- [X] recover from failed test: https://github.com/afet-yonetim-sistemi/ays-be/actions/runs/5405701616/jobs/9821608956

File: src/main/java/com/ays/user/controller/UserSelfController.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * User Self Controller class for managing user's own operations.
  */
 @RestController
-@RequestMapping("/api/v1/user/self")
+@RequestMapping("/api/v1/user-self")
```
Comment (agitrubard) [inline src/main/java/com/ays/user/controller/UserSelfController.java:18]: Why did we change the endpoint?

File: src/main/java/com/ays/user/controller/UserController.java [hunk 2]
```diff
@@ -47,7 +47,7 @@ class UserController {
      * @param listRequest The request object containing the list criteria.
      * @return A response object containing a paginated list of users.
      */
-    @PostMapping
+    @GetMapping
```
Comment (agitrubard) [inline src/main/java/com/ays/user/controller/UserController.java:50]: Can we revert this since it was edited in a different branch?

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/main/java/com/ays/user/controller/UserController.java (hunk 2 above)
Comment (skayikci) [reply src/main/java/com/ays/user/controller/UserController.java:50]: i reverted.

File: src/main/java/com/ays/user/controller/UserSelfController.java (hunk 1 above)
Comment (skayikci) [reply src/main/java/com/ays/user/controller/UserSelfController.java:18]: gives the impression that this and user controller are exactly the same, looks like even the service layer is different, so maybe create a distinct endpoint for this?

**Review by agitrubard — APPROVED** [review]

File: .github/workflows/branch-reporter.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: Maven Build
+
+on:
+  pull_request:
+    types:
+      - opened
+      - review_requested
+      - edited
+      - synchronize
```
Comment (Rapter1990) [inline .github/workflows/branch-reporter.yml:9]: @skayikci Is it possible to define `assigned `label in the types of pull request?

Here is the sample code shown below

```
pull_request:
  types:
    - assigned 
```

Comment (skayikci) [reply .github/workflows/branch-reporter.yml:9]: done.

**Comment by agitrubard** [comment]
@skayikci There are conflicts, so we cannot merge.

**Comment by skayikci** [comment]
@agitrubard resolved conflicts, please have a look.

**Review by Rapter1990 — APPROVED** [review]

**Pull request merged**